workers/unity/Packages/io.improbable.gdk.tools/.CodeGenDocs/0-overview.md

```diff
@@ -0,0 +1,83 @@
+
+This script also exposes the `Generate code` and `Generate code (force)` options under the `SpatialOS` menu in the Unity Editor.
+
+### Setup project
+
+When opening up your project in Unity, the contents of `.CodeGenTemplate/` are first copied from the `io.improbable.gdk.tools` package into the `<projectroot>/build/codegen/` directory. This ensures that the base `CodeGen` template and the `CodeGenerationLib` it depends on are ready for consumption in the project's build directory.
+
+Once this is done, the GDK generates run configurations for the code generator. This step searches all project packages for `.schema` directories, combines it with the schema source directories defined the GDK tools configuration, and sets this information in the run configurations for the code generator to use.
+
+These configurations are used to ensure the code generator is always* called with the correct, validated arguments across Visual Studio, JetBrains Rider, and the dotnet CLI.
+
+<sup>_\*unless a user tinkers with things they shouldn't be._</sup>
+
+> Note: this enables you to iterate on code generation modules within an IDE without having to switch back and forth with the Unity Editor.
+
+### Generate code
+
+When you click the `Generate code` button, the GDK:
+
+1. Finds all `.codegen` directories in the packages your Unity project depends on.
+1. Updates the build directory's `CodeGen.csproj` with references to all the codegen units that were found.
+1. Calls `dotnet run` on the updated code generator.
+
+> All standard output from running the code generator is processed and logged at the appropriate level in the Unity Editor.
+
+#### Generate code (force)
+
+The `Generate code (force)` option effectively "resets" the code generator before running it. The directory containing generated code and the previously copied code generator template are both deleted. The code generator template in the `io.improbable.gdk.tools` package is then copied across again before a freshly set up code generator is run.
+
+## `CodeGen`
+
+The base `CodeGen` template is the entry point for the code generator, and therefore responsible for:
+
+* Parsing command line arguments.
+* Invoking the schema compiler to produce a schema bundle.
+* Using `CodeGenerationLib` to parse and store important schema details.
+* Finding code generation jobs in linked codegen units.
+* Providing schema details to each code generation job it runs.
+
+Once copied into the project build directory, `CodeGen.csproj` is updated with references to any modular code generation units found in the packages that your Unity project depends on.
+
+The code generator will run all code generation jobs defined in each referenced modular codegen unit.
+
+## `CodeGenerationLib`
+
+The `CodeGenerationLib` project is a library used by the base `CodeGen` project and referenced codegen units to:
```

Comment:
```suggestion
The `CodeGenerationLib` project is a library used by the base `CodeGen` project and referenced by codegen units to:
```
?

Reply:
I was trying to say it's used by base project + the units it holds references to